bandit/core/metrics.py

```diff
@@ -30,21 +35,41 @@ def begin(self, fname):
         """Begin a new metric block.
 
         This starts a new metric collection name "fname" and makes is active.
-
         :param fname: the metrics unique name, normally the file name.
         """
-        self.data[fname] = {"loc": 0, "nosec": 0}
+        self.data[fname] = {
+            "loc": 0,
+            "nosec": 0,
+            "nosec_by_test": 0,
+            "failed_nosec_by_test": 0,
+        }
         self.current = self.data[fname]
 
     def note_nosec(self, num=1):
-        """Note a "nosec" commnet.
+        """Note a "nosec" comment.
 
         Increment the currently active metrics nosec count.
-
         :param num: number of nosecs seen, defaults to 1
         """
         self.current["nosec"] += num
 
+    def note_nosec_by_test(self, num=1):
+        """Note a "nosec BXXX, BYYY, ..." comment.
+
+        Increment the currently active metrics nosec_by_test count.
+        :param num: number of nosecs seen, defaults to 1
+        """
+        self.current["nosec_by_test"] += num
+
+    def note_failed_nosec_by_test(self, num=1):
+        """Note a test failed when using a nosec comment with specific tests.
+
+        e.g. # nosec B102, B607, but B602 failed.
+        Increment the currently active metrics failed_nosec_by_test count.
+        :param num: number of failed nosecs seen, defaults to 1
+        """
+        self.current["failed_nosec_by_test"] += num
```

Comment:
I'm still struggling to understand the value of tracking this. Can you share how you expect people to understand/use/interpret this? I get _when_ we increment it, I don't understand why this is helpful to track

Reply:
I'll remove it.